pkg/sloop/server/internal/config/config.go

```diff
@@ -75,77 +75,130 @@ type SloopConfig struct {
+	fs.BoolVar(&config.EnableDeleteKeys, "enable-delete-keys", config.EnableDeleteKeys, "Use delete prefixes instead of dropPrefix for GC")
+	fs.BoolVar(&config.BadgerVLogFileIOMapping, "badger-vlog-fileIO-mapping", config.BadgerVLogFileIOMapping, "Indicates which file loading mode should be used for the value log data, in memory constrained environments the value is recommended to be true")
+	fs.BoolVar(&config.BadgerVLogTruncate, "badger-vlog-truncate", config.BadgerVLogTruncate, "Truncate value log if badger db offset is different from badger db size")
+	return config
 }
 
+func registerDefaultSloopConfig() *SloopConfig{
+	defaultConfig:=SloopConfig{
+		ConfigFile:               "",
+		DisableKubeWatcher:       false,
+		KubeWatchResyncInterval:  30*time.Minute,
+		WebFilesPath:             "./pkg/sloop/webserver/webfiles",
+		BindAddress:              "",
+		Port:                     8080,
+		StoreRoot:                "./data",
+		MaxLookback:              time.Duration(14*24)*time.Hour,
+		MaxDiskMb:                32*1024,
+		DebugPlaybackFile:        "",
+		DebugRecordFile:          "",
+		DeletionBatchSize:        1000,
+		UseMockBadger:            false,
+		DisableStoreManager:      false,
+		CleanupFrequency:         time.Minute*30,
+		KeepMinorNodeUpdates:     false,
+		DefaultNamespace:         "default",
+		DefaultKind:              "_all",
+		DefaultLookback:          "1h",
+		UseKubeContext:           "",
+		DisplayContext:           "",
+		ApiServerHost:            "",
+		WatchCrds:                true,
+		CrdRefreshInterval:       time.Duration(5*time.Minute),
+		ThresholdForGC:           0.8,
+		RestoreDatabaseFile:      "",
+		BadgerDiscardRatio:       0.99,
+		BadgerVLogGCFreq:         time.Minute*1,
+		BadgerMaxTableSize:       0,
+		BadgerLevelOneSize:       0,
+		BadgerLevSizeMultiplier:  0,
+		BadgerKeepL0InMemory:    true,
+		BadgerVLogFileSize:       0,
+		BadgerVLogMaxEntries:     200000,
+		BadgerUseLSMOnlyOptions:  true,
+		BadgerEnableEventLogging: false,
+		BadgerNumOfCompactors:    0,
+		BadgerNumL0Tables:        0,
+		BadgerNumL0TablesStall:   0,
+		BadgerSyncWrites:         true,
+		BadgerVLogFileIOMapping:  false,
+		BadgerVLogTruncate:       true,
+		EnableDeleteKeys:         false,
+	}
+	return &defaultConfig
+}
 // This will first check if a config file is specified on cmd line using a temporary flagSet
 // If not there, check the environment variable
 // If we have a config path, load initial values from it
 // Next parse flags again and override any fields from command line
 //
 // We do this to support settings that can come from either cmd line or config file
 func Init() *SloopConfig {
-	newConfig := &SloopConfig{}
+	finalConfig := registerDefaultSloopConfig()
+
+	configFileName:=""
+	configFileFlag := preParseConfigFlag()
+	configFileOS := os.Getenv(sloopConfigEnvVar)
```

Comment:
it's for the logging purposes, wanted to log if we are using the env var or the flag passed. It adds clarification and answers which config file we prioritize, in case both are set.

Reply:
with the snippet above we can also do the same logging, can't we?

Reply:
Reusing variables adds an element of risk, making the code harder to follow and more error prone.

Reply:
IMO, right now the code is less readable than snippet above, harder to follow. This adds complexity and can also be error prone.

Reply:
I don't agree it's less readable currently. but we can get more opinions, see what others think

Reply:
Ok, so two things:
1) Yes, you can definitely add the logs to the snippet I proposed, no worries there. Here's what that can look like:

``` bash
  # Set config file name
	configFileName :=  preParseConfigFlag()
	if configFileName == ""  {
		configFileName := os.Getenv(sloopConfigEnvVar)
                glog.Infof("Set via environmental variables: %s", configFileName)
	} else {
               glog.Infof("Set via flag: %s", configFileName)
        }

# Load config file
        if configFileName != "" {
                 finalConfig = loadFromFile(configFileName, finalConfig)
        }
```

2) I am with nurland on this - I **do not want** to know about the different ways of setting the configFileName - all that we need, is a way to determine how to set it. Once set - we use it. That makes the proposed code snippet easier to read. 

With multiple variables, now I need to know **which** ones are set, **why** they are set, and **what** I need to do with them - individually. The proposed snippet says - set the config file name either first from the flags, or second, from the env vars, then load it.

Reply:
I don't understand why both of u are against defining variables? I don't think it is harming either the functionality or the readability. The implementation currently is basically "lets get all the information on the table and then decide which to use" rather than "let me get some information and see if that can be used, if not let me try something else". 
The only actual difference I see between current and proposed implementation, is **two** variables.

Reply:
I think my main point is that _for me_ the current code was much harder to follow than snippet above. Seems like it was hard to follow for @garakly as well. Having said that I don't feel strongly about, not a PR blocker for me.

Reply:
makes sense! let me try to make it more readable.